clifford/tools/g3c/__init__.py

```diff
@@ -428,10 +431,74 @@ def closest_points_on_circles(C1, C2, niterations=20):
     return P1, P2
 
 
-def closest_points_circle_line(C, L, niterations=20):
+def closest_point_on_line_from_circle(C, L):
+    """
+    Returns the point on the line L that is closest to the circle C
+    Uses the algorithm described in Appendix A of Andreas Aristidou's PhD thesis
+    """
+    return project_points_to_line([closest_point_on_circle_from_line(C, L)], L)[0]
+
+
+def closest_point_on_circle_from_line(C, L):
+    """
+    Returns the point on the circle C that is closest to the line L
+    Uses the algorithm described in Appendix A of Andreas Aristidou's PhD thesis
+    """
+    phi = (C^einf).normal()
+    B = meet(L, phi)
+    A = normalise_n_minus_1((C * einf * C)(1))
+    bound_sphere = ((C * phi) * I5).normal()
+    if abs((B**2)[0]) < 1E-6:
+        # The line and plane of the circle are parallel
+        # Project the line into the plane
+        Lpln = (L.normal() + (phi*L*phi)(3).normal()).normal()
+        # Project the centre of the circle onto the line
+        X = (A|Lpln)*einf*(A|Lpln)
+        if sphere_in_sphere(X*I5, bound_sphere):
+            # The circle and line intersect
+            PP = meet(Lpln, bound_sphere)
+            return point_pair_to_end_points(PP)[0]
+        else:
+            L2 = A ^ X ^ einf
+            PP = meet(L2, bound_sphere)
+            return point_pair_to_end_points(PP)[1]
```

Comment:
Codecov says these lines are not tested. Can you construct a test by hand that hits here, by making the plane and circle coplanar?